### ISSUE TITLE
Bug : all attributes must be indexed

### ISSUE THREAD

**Issue by imdkbj** [sst/sst#4163]
This is not mandatory to index all attributes, Getting subjected error for the following code.

```js
const _stage = $app.stage

export const impersonateTable = new sst.aws.Dynamo('DynamoImpersonate', {
  fields: {
    id: 'string',
    createdAt: 'number'
  },
  primaryIndex: { hashKey: 'id' },
  transform: {
    table: {
      billingMode: 'PAY_PER_REQUEST',
      deletionProtectionEnabled: true,
      name: `${_stage}-sso-impersonate`,
      pointInTimeRecovery: {
        enabled: false
      }
    }
  }
})
```

**Comment by jdlms** [comment]
I am also getting this error.

` sdk-v2/provider2.go:520: sdk.helper_schema: all attributes must be indexed. Unused attributes: ["img" "lastUpdated" "price"]
: provider=aws@6.56.1`

**Comment by fructusmortus** [comment]
I'm getting the same error. Do you have plans to fix it in the nearest future? Thx

**Comment by jayair** [comment]
We must've missed this.

@fwang let's take a look?

**Comment by fwang** [comment]
I don't think this is a bug. You get this error if you include attributes in `fields` that are not used in indexes.

@imdkbj try removing `createdAt: 'number'` from `fields`.

@jdlms try removing `img`, `lastUpdated`, and `price` from the `fields` list.

@fructusmortus can i see how u r defining the table?

**Comment by fructusmortus** [comment]
@fwang Yes, I know that it is possible, and I have already done it in my project. It's just that in the previous version, it was possible to define fields that were not indexes, and that was convenient. Is it possible to restore this in v3? Thanks!

**Comment by jayair** [comment]
@fructusmortus are you talking about SST v2?

**Comment by fructusmortus** [comment]
@jayair Yes, by 'previous version,' I mean SST v2.

**Comment by Born-Winga** [comment]
Having the same issue.

**Comment by fwang** [comment]
@OYOOOWINO can i see how the table is defined?

**Comment by fwang** [comment]
@fructusmortus hmm... can I see your code just to make sure i understand what you are looking to do?

**Comment by fructusmortus** [comment]
@fwang For example I have the following v3 table: 
```
const usersTable = new sst.aws.Dynamo('Users', {
    fields: {
        userId: 'string',
        // email: 'string',
        // name: 'string',
        // familyName: 'string',
        // avatar: 'string',
    },
    primaryIndex: {
        hashKey: 'userId',
    }
})
```
I commented out all the fields that aren't indexes to work around the problem. In version 2, it was possible to keep all these fields without them being indexes, like this:
```
const usersTable = new Table(stack, 'Users', {
        fields: {
            userId: 'string',
            email: 'string',
            name: 'string',
            familyName: 'string',
            avatar: 'string',
        },
        primaryIndex: {
            partitionKey: 'userId',
        }
    })
```

**Comment by afitterling** [comment]
Hi there, I am currently using SST 3.10.0 and I am still having the above problem? Is there any example with more then two attributes available in the Repo?
Is this solved?

**Comment by imdkbj** [comment]
Hey @fwang , Sorry for the missing reply here

Yes. We only need to add columns that are used in the index in the config. Even this is mentioned in the documentation.
